classes/PodsRESTHandlers.php

```diff
@@ -59,58 +59,64 @@ public static function get_handler( $object, $field_name, $request ) {
 		$id       = pods_v( 'id', $object );
 		$pod      = self::get_pod( $pod_name, $id );
 		if ( $pod && PodsRESTFields::field_allowed_to_extend( $field_name, $pod, 'read' ) ) {
+
 			$params     = null;
 			$field_data = $pod->fields( $field_name );
 			if ( 'pick' == pods_v( 'type', $field_data ) ) {
+
 				$output_type = pods_v( 'rest_pick_response', $field_data['options'], 'array' );
 				if ( 'array' == $output_type ) {
 
 					$related_pod = $pod->field( $field_name, array( 'output' => 'pod' ) );
-					$fields      = $related_pod->fields();
-					$fields      = array_keys( $fields );
-					if ( isset( $related_pod->pod_data['object_fields'] ) && ! empty( $related_pod->pod_data['object_fields'] ) ) {
-						$fields = array_merge( $fields, array_keys( $related_pod->pod_data['object_fields'] ) );
-					}
+					if ( is_a( $related_pod, 'Pods' ) ) {
```

Comment:
@sc0ttkclark To clarify $related_pod is 1 Pods object `$related_pod = $pod->field( $field_name, array( 'output' => 'pod' ) );` I don't see anyway Pods could return an array there.

@pglewis - I don't understand the need for `checking if the related object is actually exposed to the REST API.` IF user didn't want to query form this Pod to related Pod user would not have enabled it in UI.

I think best solution is after `$fields      = $related_pod->fields();` if $fields is false, make it into an  empty array, that way fields can be added to it using or it will return empty `pods_rest_api_fields_for_relationship_response` fitler.